### DISCUSSION TITLE
What is the state management do you recommend?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [blitz-js/blitz#3081] [Q&A]
Hi, I am trying to choose a state management tool, what tool do you suggest ? Recoil, Zudtand or Redux?

**Comment by abuuzayr** [comment]
Depends on your needs, really. Tiny use case? The context API is enough. Need more? Recoil should do the job. [Redux is not usually recommended as a starting point nowadays](https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367)  with so many alternatives, but it can do the job.

**Reply by IRediTOTO** [reply]
I just wonder, Blitz is using react-query, can `react-query` use like a management state? If not i think I will choose recoil

**Reply by JohannesKlauss** [reply]
I would ask myself if I really need a state management library. Most data you handle is just data from the server. Most of the time there is no need to duplicate that data into a recoil oder redux store. react-query has you covered there. If a web app actually creates local data that stays local (like a dark mode switch) then you can use a state management library if it gets complex. Otherwise context API oder local storage is more than sufficient.
